pkg/app/run.go

```diff
@@ -73,7 +73,14 @@ func (r *Run) withPreparedCharts(helmfileCommand string, opts state.ChartPrepare
 
 	concurrency := opts.Concurrency
 
-	releaseToChart, errs := r.state.PrepareCharts(r.helm, dir, concurrency, helmfileCommand, opts)
+	var releaseToChart map[state.PrepareChartKey]string
+	var errs []error
+
+	if strings.EqualFold(helmfileCommand, "write-values") || strings.EqualFold(helmfileCommand, "list") {
+		// Skip chart preparation for local commands - write-values / list
+	} else {
+		releaseToChart, errs = r.state.PrepareCharts(r.helm, dir, concurrency, helmfileCommand, opts)
+	}
```

Comment:
could you move the logic to an sigle func? so we can add some tests easily.

Reply:
Sure, give me some time and I'll fix this and add some tests

Reply:
@uristernik thanks so much.